component/src/main/java/com/celements/photo/utilities/DefaultImageUrlExtractor.java

```diff
@@ -0,0 +1,119 @@
+  private final Pattern EXTRACT_SPACE_PATTERN = Pattern.compile("^(.*?/){2}(.*?)/.*$");
+  private final Pattern EXTRACT_DOCUMENT_PATTERN = Pattern.compile("^(.*?/){3}(.*?)/.*$");
+  private final Pattern EXTRACT_FILENAME_PATTERN = Pattern.compile("^(.*?/){4}(.*?)(|\\?.*)$");
+
+  @Override
+  public @NotNull List<ImageUrlDim> extractImagesList(@NotNull String content) {
+    Map<Long, List<ImageUrlDim>> articleImages = new TreeMap<>();
+    Matcher m = IMG_FROM_HTML_PATTERN.matcher(content);
+    while (m.find()) {
+      String imgUrl = m.group(1);
+      Long key = getImgUrlSizeKey(imgUrl);
+      if (!articleImages.containsKey(key)) {
+        articleImages.put(key, new ArrayList<ImageUrlDim>());
+      }
+      articleImages.get(key).add(getImgUrlExternal(imgUrl));
+    }
+    return filterImagesByMissingOrSmallSize(articleImages);
+  }
+
+  List<ImageUrlDim> filterImagesByMissingOrSmallSize(Map<Long, List<ImageUrlDim>> articleImages) {
+    List<ImageUrlDim> sortedImages = new ArrayList<>();
+    for (Long imgArea : articleImages.keySet()) {
+      // Image size not extractable is in key == '-1' Don't include too small images.
+      if ((imgArea == -1) || (imgArea >= (MIN_SOCIAL_MEDIA_AREA_SIZE))) {
+        Collections.reverse(articleImages.get(imgArea));
+        sortedImages.addAll(articleImages.get(imgArea));
+      }
+    }
+    return sortedImages;
+  }
+
+  ImageUrlDim getImgUrlExternal(String imgUrl) {
+    if (!imgUrl.startsWith("http://") && !imgUrl.startsWith("https://")) {
+      String action = getMatchedPart(EXTRACT_ACTION_PATTERN.matcher(imgUrl), 1);
+      String space = getMatchedPart(EXTRACT_SPACE_PATTERN.matcher(imgUrl), 2);
+      String docname = getMatchedPart(EXTRACT_DOCUMENT_PATTERN.matcher(imgUrl), 2);
+      String filename = getMatchedPart(EXTRACT_FILENAME_PATTERN.matcher(imgUrl), 2);
+      String query = imgUrl.substring(Math.min(imgUrl.indexOf('?') + 1, imgUrl.length()));
+      XWikiContext context = Utils.getComponent(ModelContext.class).getXWikiContext();
+      imgUrl = context.getURLFactory().createAttachmentURL(filename, space, docname, action, query,
+          context.getDatabase(), context).toString();
+    }
+    return new ImageUrlDim(imgUrl, parseImgUrlDimension(imgUrl, "celwidth"), parseImgUrlDimension(
+        imgUrl, "celheight"));
+  }
+
+  String getMatchedPart(Matcher m, int group) {
+    if (m.find()) {
+      return m.group(group);
+    }
+    return "";
+  }
+
+  Long getImgUrlSizeKey(String imgUrl) {
+    long w = parseImgUrlDimension(imgUrl, "celwidth");
+    long h = parseImgUrlDimension(imgUrl, "celheight");
+    Long area = new Long(h * w);
+    if (area < 0) {
+      area = area * area;
+    } else if ((h == -1) && (w == -1)) {
+      area = -1l;
+    }
+    return area;
+  }
+
+  int parseImgUrlDimension(String imgUrl, String dimension) {
+    String str = imgUrl.replaceAll("^.*" + dimension + "=(\\d+)(&.*)?$", "$1");
```

Comment:
You only use two different regexes. Please use static regexes and pattern. Change dimension parameter to the regex pattern constant.